pkg/recipes/terraform/module.go

```diff
@@ -98,13 +99,8 @@ func inspectModule(workingDir, localModuleName string) (*moduleInspectResult, er
 // downloadModule downloads the module to the workingDir from the module source specified in the Terraform configuration.
 // It uses Terraform's Get command to download the module using the Terraform executable available at execPath.
 // An error is returned if the module could not be downloaded.
-func downloadModule(ctx context.Context, workingDir, execPath, templatePath string) error {
-	tf, err := NewTerraform(ctx, workingDir, execPath)
-	if err != nil {
-		return err
-	}
-
-	if err = tf.Get(ctx); err != nil {
+func downloadModule(ctx context.Context, tf *tfexec.Terraform, templatePath string) error {
```

Comment:
I think that, in the future, we could have a struct that has **tfexec.Terraform** as a property. That could have a constructor and some functions. One of the functions could be **downloadModule**, which could be something like this:

```
func (td *TerraformDriver) downloadModule(ctx context.Context, templatePath string) error {
         td.Terraform.Get(ctx)...
...
}
```

Note: The struct name will probably something other than TerraformDriver :)

Reply:
This way we could also mock some of the functions of **tfexec.Terraform**.

Reply:
That makes sense to me. Some of the functions called from execute.go can be cleaned up as well downloadModule, inspectModule etc, but it's best to do that in smaller PRs so didn't include that here.